server/service/packs.go

```diff
@@ -30,19 +31,35 @@ func packResponseForPack(ctx context.Context, svc fleet.Service, pack fleet.Pack
 		return nil, err
 	}
 
+	totalHostsCount := uint(0)
+
 	hostMetrics, err := svc.CountHostsInTargets(
 		ctx,
 		nil,
-		fleet.HostTargets{HostIDs: pack.HostIDs, LabelIDs: pack.LabelIDs, TeamIDs: pack.TeamIDs},
+		fleet.HostTargets{
+			HostIDs:  pack.HostIDs,
+			LabelIDs: pack.LabelIDs,
+			TeamIDs:  pack.TeamIDs,
+		},
 	)
 	if err != nil {
-		return nil, err
+		// Some users (e.g. gitops) are not able to read targets, thus
+		// we do not fail when gathering the total host count to not fail
+		// the request.
```

Comment:
Question: If it fails for one of the legit roles, should we not fail here?

Reply:
e.g. if the read fails for global admin.

Reply:
Maybe your rule should be:
```
If ( GitOs AND err != nil ) {
    don't fail
} else {
     fail 
}

```

?

Reply:
Agree. Even if it doesn't look like other roles will hit this, we need to make this future proof in case we add other users with reduced permissions like GitOps. 1bad45bce.